GNUmakefile

```diff
@@ -16,11 +16,32 @@ ifneq ($(origin TRAVIS_TAG), undefined)
 	VERSION := $(TRAVIS_TAG)
 endif
 
+# For changelog generation, default the last release to the last tag on
+# any branch, and this release to just be the current branch we're on.
+LAST_RELEASE?=$$(git describe --tags $$(git rev-list --tags --max-count=1))
+THIS_RELEASE?=$$(git rev-parse --abbrev-ref HEAD)
+
 default: build
 
 build: fmtcheck
 	go build -v .
 
+# expected to be invoked by make gen/changelog LAST_RELEASE=gitref THIS_RELEASE=gitref
+.PHONY: changelog-build
```

Comment:
There is a `.PHONY` entry at the bottom of the file where you can add the target names. 

As a side note, the whole `PHONY` thing is because Makefiles were originally used for building C files and so targets were the names of actual files – anything that wasn't a file was a "phony" target.